apps/web/src/lib/auth/getSession.ts

```diff
@@ -0,0 +1,46 @@
+import { lucia } from '$/lib/auth'
+import type { Session, User } from 'lucia'
+
+export async function getSession(): Promise<
+  { user: User; session: Session } | { user: null; session: null }
+> {
+  // NOTE: We dynamically import the cookies function to make Nextjs happy
+  // Info: https://github.com/vercel/next.js/issues/49757
+  const { cookies } = await import('next/headers')
+  const sessionId = cookies().get(lucia.sessionCookieName)?.value ?? null
+
+  if (!sessionId) {
+    return {
+      user: null,
+      session: null,
+    }
+  }
+
+  const result = await lucia.validateSession(sessionId)
+
+  try {
+    if (result.session && result.session.fresh) {
+      const sessionCookie = lucia.createSessionCookie(result.session.id)
+      cookies().set(
+        sessionCookie.name,
+        sessionCookie.value,
+        sessionCookie.attributes,
+      )
+    }
```

Comment:
Explain to me why do we need to set the session cookie again if it has already been validated in line 19?

Reply:
I followed [lucia docs](https://lucia-auth.com/tutorials/username-and-password/nextjs-app)
![image](https://github.com/user-attachments/assets/97b86672-c7d6-4314-bf0b-c14fe68c2f56)